src/libmongoc/tests/test-mongoc-cluster.c

```diff
@@ -1813,12 +1811,114 @@ _test_cmd_on_unknown_serverid (bool pooled)
 }
 
 void
-test_cmd_on_unknown_serverid (void)
+test_cmd_on_unknown_serverid_pooled (void)
 {
-   _test_cmd_on_unknown_serverid (false /* pooled */);
    _test_cmd_on_unknown_serverid (true /* pooled */);
 }
 
+void
+test_cmd_on_unknown_serverid_single (void)
+{
+   _test_cmd_on_unknown_serverid (false /* pooled */);
+}
+
+
+/* Test that server streams are invalidated as expected. */
+static void
+test_cluster_stream_invalidation_single (void)
+{
+   mongoc_client_t *client;
+   mongoc_server_description_t *sd;
+   bson_error_t error;
+   mongoc_server_stream_t *stream;
+
+   client = test_framework_new_default_client ();
+   /* Select a server to start monitoring. */
+   sd = mongoc_client_select_server (
+      client, true /* for writes */, NULL /* read prefs */, &error);
+   ASSERT_OR_PRINT (sd, error);
+
+   /* Test "clearing the pool". This should invalidate existing server streams.
+    */
+   stream = mongoc_cluster_stream_for_writes (
+      &client->cluster, NULL /* session */, NULL /* reply */, &error);
+   ASSERT_OR_PRINT (stream, error);
+   BSON_ASSERT (mongoc_cluster_stream_valid (&client->cluster, stream));
+   _mongoc_topology_clear_connection_pool (client->topology,
```

Comment:
It's suitable to call this method in single-threaded mode? I suppose it doesn't care about how many streams are actually present and just wipes everything?

Reply:
Yes, it still invalidates the one stream. Though it is not particularly useful in single-threaded clients since there is only one stream.